### DISCUSSION TITLE
Checkbox value inconsistent

### DISCUSSION THREAD

**Discussion by jira-zz** [react-hook-form/react-hook-form#11351] [Q&A] [answered]
I'm using RHF with zod, validating form with checkbox.
I observed a weird behavior: when the checkbox is clicked the value passed to zod schema is string, but when the checkbox is not checked the value is boolean (false).
Is that an expected behavior?

**Comment by Moshyfawn** [comment]
Unable to reproduce. Please take your time to modify [this codesandbox](https://codesandbox.io/s/elastic-hill-67wqxm?file=%2Fsrc%2FApp.tsx) to display the behavior you are describing.

**Reply by jira-zz** [reply, marked as answer]
The issue was I used _defaultValue_ instead of _defaultChecked_ on the checkbox. After fixing that it works as expected.

Thanks

**Reply by Moshyfawn** [reply]
See, we couldn't have figured that out without a reproduction sandbox 😉

I'm glad you were able to solve it.
